src/main/java/io/strimzi/kafka/bridge/http/HttpSinkBridgeEndpoint.java

```diff
@@ -188,6 +194,53 @@ public void handle(Endpoint<?> endpoint) {
                     sendConsumerCommitOffsetResponse(routingContext.response(), status.succeeded());
                 });
                 break;
+
+            case SEEK:
+
+                JsonArray seekOffsetsList = bodyAsJson.getJsonArray("offsets");
+
+                List<Future> seekHandlers = new ArrayList<>(seekOffsetsList.size());
+                for (int i = 0; i < seekOffsetsList.size(); i++) {
+                    TopicPartition topicPartition = new TopicPartition(seekOffsetsList.getJsonObject(i));
+                    long offset = seekOffsetsList.getJsonObject(i).getLong("offset");
+                    Future<Void> fut = Future.future();
+                    seekHandlers.add(fut);
+                    this.seek(topicPartition, offset, fut.completer());
+                }
+
+                CompositeFuture.join(seekHandlers).setHandler(done -> {
+                    if (done.succeeded()) {
+                        sendSeekResponse(routingContext.response(), ErrorCodeEnum.NO_CONTENT);
+                    } else {
+                        sendSeekResponse(routingContext.response(), ErrorCodeEnum.INTERNAL_SERVER_ERROR);
+                    }
+                });
+                break;
+
+            case SEEK_TO_BEGINNING:
+            case SEEK_TO_END:
+
+                JsonArray seekPartitionsList = bodyAsJson.getJsonArray("partitions");
+
+                Set<TopicPartition> set = seekPartitionsList.stream()
+                        .map(JsonObject.class::cast)
+                        .map(json -> new TopicPartition(json.getString("topic"), json.getInteger("partition")))
+                        .collect(Collectors.toSet());
```

Comment:
So the API contract on duplicates is last-in-wins?

Reply:
For seek beginning/end you are not specifying an offset but just topic + partition on which you want to seek at beginning or end (remember that the beginning and end endpoint are different).
So if you write the same topic + partition in the JSON you are just duplicating the same information, there is no last-in-wins. Or I miss understood the question?

Reply:
Excellent point sir!